File: app-java/src/main/java/science/monke/internal/order/repository/OrderRepository.java
package science.monke.internal.order.repository;

import org.springframework.data.repository.CrudRepository;
import science.monke.internal.order.entity.OrderEntity;

import java.util.Optional;
import java.util.UUID;

public interface OrderRepository extends CrudRepository<OrderEntity, Long> {
  Optional<OrderEntity> findByOrderId(UUID orderId);
}

File: app-java/src/main/kotlin/science/monke/internal/order/repository/OrderRepository.kt
package science.monke.internal.order.repository;

import org.springframework.data.repository.CrudRepository
import science.monke.internal.order.entity.OrderEntity
import java.util.*

interface OrderRepository : CrudRepository<OrderEntity, Long> {
    fun findByOrderId(orderId: UUID): Optional<OrderEntity>
}
